Rename save_contact to create_contact

lloydnye opened the pull request.

Commit:
Rename save_contact to create_contact

frankoid commented:
I'm planning to rename the simulators and tests to match and then merge this...

Commit:
More save/create contact renames

Change save contact to create contact in test and simulator classes.

Rename SaveContactClientError to UnsuccessfulCreateContact to match the Unsuccessful prefix that the existing exception-throwing simulators use.

frankoid commented:
I've made the changes and also renamed SaveContactClientError to UnsuccessfulCreateContact.
OK with you to merge?
